scripts/ocr_runner.py

```
@@ -0,0 +1,198 @@
+
+def run_ocrmypdf(input_pdf, out_pdf, log_path, timeout):
+    with open(log_path, 'wb') as lf:
+        proc = subprocess.run(['ocrmypdf', '--deskew', '--force', '--output-type', 'pdf', str(input_pdf), str(out_pdf)], stdout=lf, stderr=lf, timeout=timeout)
+    return proc.returncode
+
+
+def fallback_tesseract(input_pdf, out_txt_path):
+    # use pdftoppm -> tesseract
+    tmp_prefix = f'/tmp/{out_txt_path.stem}_page'
+    try:
+        subprocess.run(['pdftoppm', '-r', '300', str(input_pdf), tmp_prefix], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, timeout=180)
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)